### PR TITLE
Add python tests for various Dtypes

### PR DESCRIPTION
**Pull request by nickgg**
A few python tests to cover the Tensor interface for Dtypes.

### CODE REVIEW

File: test/test_tensorexpr.py [hunk 1]
```diff
@@ -376,6 +376,65 @@ def easy(x, y):
...
+        )
+
+        a = torch.rand(1024, dtype=torch.double)
+        b = torch.full((1024,), 0.5, dtype=torch.double)
+        x = traced(a, b)
+        np.testing.assert_allclose(a.numpy() + b.numpy(), x.numpy())
+
+    def test_short(self):
+        def easy(x, y):
+            aaa = torch.add(x, y)
```
Comment (bertmaher) [inline test/test_tensorexpr.py:399]: I'm not sure the fuser gets invoked if there's just one `aten` op in the region.  You might need to do something like `x * y * y`.

File: test/test_tensorexpr.py [hunk 2]
```diff
@@ -376,6 +376,65 @@ def easy(x, y):
...
         np.testing.assert_allclose(a.numpy() + b.numpy(), x.numpy())
 
+    def test_double(self):
+        def easy(x, y):
+            aaa = torch.add(x, y)
+            return aaa
+
+        traced = torch.jit.trace(
+            easy,
+            (torch.rand(1024, dtype=torch.float64), torch.full((1024,), 0.5, dtype=torch.float64)),
```
Comment (bertmaher) [inline test/test_tensorexpr.py]: I think I set the precedent of using 1024 here but it's overkill and makes our tests kind of slow.  Let's use something small like 4.

**Review by bertmaher — APPROVED** [review]
Aside from those minor things, this looks good.  Fire when ready :)

**Comment by nickgg** [comment]
@bertmaher 👍

**Pull request merged**